src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt

```diff
@@ -173,16 +175,46 @@ data class SentenceCalculation(
   }
 
   val earlyReleaseSchemeEligibilityDate: LocalDate? get() {
-    if (numberOfDaysToErsed == null) {
-      return null
+    if (calculateErsed) {
+      if (sentence.identificationTrack.calculateErsedFromHalfway()) {
+        return calculateErsedFromHalfway()
+      }
+      if (sentence.identificationTrack.calculateErsedFromTwoThirds()) {
+        return calculateErsedFromTwoThirds()
+      }
     }
-    return sentence.sentencedAt
-      .plusDays(numberOfDaysToErsed.toLong())
-      .plusDays(calculatedTotalAddedDays.toLong())
-      .minusDays(calculatedTotalDeductedDays.toLong())
-      .plusDays(calculatedTotalAwardedDays.toLong())
+    return null
   }
 
+  private fun calculateErsedFromTwoThirds(): LocalDate {
+    val custodialDuration = sentence.custodialDuration()
+    val days = custodialDuration.getLengthInDays(sentence.sentencedAt)
+    return if (days >= 1097) {
```

Comment:
that's an odd rule >= 1097 days! can we stick the 1097 into a named constant to provide context, presumably it represents some policy value?

Reply:
1097 is roughly 3 years, 1463 is roughly 4 years. It's an exact number of days rather than saying years, which could be a different number of days because of leap years!